### PR TITLE
Hot fix

### PR DESCRIPTION
**Pull request by TApplencourt**
Fix:
 - metadata
 - `env_fetch` for whatever reason what broken. Was return `is_mpi_env?` true and then false...

### CODE REVIEW

File: xprof/xprof.rb.in [hunk 1]
```diff
@@ -565,16 +571,16 @@ def print_help_and_exit(parser)
...
 end
 
 # Parsing ARGV
 print_help_and_exit(parser) if ARGV.empty?
 
 begin
   parser.parse!(into: options)
-rescue OptionParser::InvalidOption, OptionParser::MissingArgument => e
-  puts(e)
+rescue  OptionParser::InvalidOption, OptionParser::MissingArgument, OptionParser::ParseError => e
```
Comment (TApplencourt) [inline xprof/xprof.rb.in:582]:
```suggestion
rescue OptionParser::InvalidOption, OptionParser::MissingArgument, OptionParser::ParseError => e
```

File: xprof/xprof.rb.in [hunk 2]
```diff
@@ -545,18 +549,20 @@ parser.on('-k', '--kernel-verbose',
...
           'Use -1 to mean no limit.',
-          'Default: 80') { |size| size.to_i }
+          'Default: 80') { |size| to_i_or_raise(size) }
 
 parser.on('--metadata', 'Display trace Metadata')
 parser.on('-v', '--version', 'Display THAPI version')
-parser.on('--debug [LEVEL]', 'Level of debug [default 0]') { |level| level.to_i }
+parser.on('-h', '--help', 'Display this message')
+
+parser.on('--debug [LEVEL]', 'Level of debug [default 0]') { |level| to_i_or_raise(level) }
```
Comment (Kerilk) [inline xprof/xprof.rb.in:558]: Why not:
```suggestion
parser.on('--debug [LEVEL]', OptionParser::DecimalInteger, 'Level of debug [default 0]')
```

File: xprof/xprof.rb.in [hunk 3]
```diff
@@ -518,13 +517,18 @@ end
 #   |_) _. ._ _ o ._   _    /  |   |
 #   |  (_| | _> | | | (_|   \_ |_ _|_
 #                      _|
+
+def to_i_or_raise(str)
+  str.match?(/^\d+$/) ? str.to_i : raise(OptionParser::ParseError.new("Invalid value (#{str}). Only integer accepted"))
+end
+
```
Comment (Kerilk) [inline xprof/xprof.rb.in:524]:
```suggestion
```
See comment below.

File: xprof/xprof.rb.in (hunk 2 above)
Comment (TApplencourt) [reply xprof/xprof.rb.in:558]: Oh, I never I didn't know. Far better! Let me try!

**Review by Kerilk — DISMISSED** [review]

**Review by Kerilk — APPROVED** [review]

**Pull request merged**